deprecated. correct db.Commit; reduce disk IO

lispc opened the pull request.

lispc committed:
done

lightsing commented:
## Preformance Review

### Result

Following tests are running on same Ubuntu server with SSD. tps is estimated with normal dist.
Bench using integration-test/bench branch
geth start cmd: https://github.com/scroll-tech/integration-test/blob/bench/scripts/start.sh#L11

### Native transaction
- zkrollup-smt estimated with 216.301 tps (13 valid samples)
- zkrollup-smt-tx estimated with 253.812 tps (+17.34%) (14 valid samples)

### ERC20 transaction
- zkrollup (HEAD): 41.247 tps (31 valid samples)
- zkrollup-smt estimated with 42.437 tps (32 valid samples)
- zkrollup-smt-tx estimated with 42.345 tps (30 valid samples)

comments by zzhang:
all the tps is ~42. So it may be related to block gas limit? not the trie performance...

lispc committed:
trie: put a sanity check in `if debug`

lispc commented:
@hsyodyssey fyi this is the current performance optimization branch.

hsyodyssey commented:
> @hsyodyssey fyi this is the current performance optimization branch.

Awsome!

noel2004 committed:
revert ApplyTransaction

revert the behavior so the PostState in receipt is consistent with the main branch geth (null after byzantium fork)
